lightning/restore/restore.go

```diff
@@ -304,48 +305,210 @@ outside:
 	return errors.Trace(err)
 }
 
-func (rc *RestoreController) restoreSchema(ctx context.Context) error {
-	if !rc.cfg.Mydumper.NoSchema {
-		if rc.tidbGlue.OwnsSQLExecutor() {
-			db, err := DBFromConfig(rc.cfg.TiDB)
-			if err != nil {
-				return errors.Annotate(err, "connect to tidb failed")
-			}
-			defer db.Close()
-			db.ExecContext(ctx, "SET SQL_MODE = ?", rc.cfg.TiDB.StrSQLMode)
-		}
+type schemaStmtType int
+
+const (
+	schemaCreateDatabase = iota
+	schemaCreateTable
+	schemaCreateView
+)
 
-		for _, dbMeta := range rc.dbMetas {
-			task := log.With(zap.String("db", dbMeta.Name)).Begin(zap.InfoLevel, "restore table schema")
+type schemaJob struct {
+	dbName string
+	stmts  []*schemaStmt
+}
+
+type schemaStmt struct {
+	tblName  string
+	stmtType schemaStmtType
+	sql      string
+}
+
+type restoreSchemaWorker struct {
+	ctx      context.Context
+	quit     context.CancelFunc
+	jobCh    chan *schemaJob
+	errCh    chan error
+	wg       sync.WaitGroup
+	glue     glue.Glue
+	store    storage.ExternalStorage
+	sessions chan checkpoints.Session
+}
 
-			tablesSchema := make(map[string]string)
+func (worker *restoreSchemaWorker) makeJobs(dbMetas []*mydump.MDDatabaseMeta) {
+	//TODO: maybe we should put these create statements into a transaction
```

Comment:
outdated TODO?

Reply:
No, this is the previous code, I just keep it. we hasn't provide transaction for now.

Reply:
we can remove it now, because a session could act partly like "put these create statements into a transaction", and DDLs can't be assembled into an atomic transaction. please check @glorv

Reply:
🙆‍♂️

Reply:
This comment was added by me? since we can't run multi ddl jobs in a txn, it should be outdated, but we can remove it later?